Source/Classes/AnimatedImages/PINAnimatedImage.m

```diff
@@ -78,24 +96,25 @@ - (NSTimeInterval)minimumFrameInterval
     return (scaledGCD / kGreatestCommonDivisorPrecision);
 }
 
-//Credit to FLAnimatedImage ( https://github.com/Flipboard/FLAnimatedImage ) for display link interval calculations
+// In practice, a recursive method should be fine because we're never looking for a really
+// large gcd.
 static NSUInteger gcd(NSUInteger a, NSUInteger b)
 {
     // http://en.wikipedia.org/wiki/Greatest_common_divisor
-    if (a < b) {
-        return gcd(b, a);
-    } else if (a == b) {
-        return b;
-    }
-    
-    while (true) {
-        NSUInteger remainder = a % b;
-        if (remainder == 0) {
-            return b;
-        }
-        a = b;
-        b = remainder;
+    NSCAssert(a > 0 && b > 0, @"A and B must be greater than 0");
+    if (a == b) {
+        return a;
+    } else if (a > b) {
+        return gcd(a - b, b);
+    } else {
+        return gcd(a, b - a);
     }
 }
 
+// Used only in testing
++ (NSUInteger)greatestCommonDevisor:(NSUInteger)a b:(NSUInteger)b
```

Comment:
Nit: It's "divisor" not "devisor".
Another nit: Might be good to restructure a bit to call out what the params are, something like `+greatestCommonDivisorBetweenInteger:andInteger:`.